src/Storages/KeyDescriptionMonotonicityChecker.cpp

```diff
@@ -0,0 +1,247 @@
+#include "KeyDescriptionMonotonicityChecker.h"
+
+#include <Functions/FunctionFactory.h>
+#include <Storages/MergeTree/KeyCondition.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+
+    void buildRPNFunctionListImpl(const RPNBuilderTreeNode & node,
```

Comment:
Adaptation of `KeyCondition::isKeyPossiblyWrappedByMonotonicFunctionsImpl`. Original code checks if the unwrapped column belongs to the key columns. This check seemed unnecessary in our case, thus I removed it and it now assumes the unwrapped column is always part of the key columns

Reply:
It also has a check for `array_joined_columns`, but honestly I don't know what's the purpose of that